[GStreamer] Move most GStreamerCommon classes implemenations to cpp file

philn opened the pull request:
#### e67dd6f289db6671b04a9f74c47dc54cd266c257
<pre>
[GStreamer] Move most GStreamerCommon classes implemenations to cpp file
<a href="https://bugs.webkit.org/show_bug.cgi?id=274524">https://bugs.webkit.org/show_bug.cgi?id=274524</a>

Reviewed by Xabier Rodriguez-Calvar.

The GStreamerCommon.h file is included in many other files, so any change there triggers ~950
compile jobs on an incremental build. The template classes were left in the header, moving them to
the cpp file would increase verbosity too much. Also driving-by a couple GstMappedFrame methods were
renamed, adopting the camelCase style.

* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::GstMappedOwnedBuffer::create):
(WebCore::GstMappedOwnedBuffer::~GstMappedOwnedBuffer):
(WebCore::GstMappedOwnedBuffer::createSharedBuffer):
(WebCore::GstMappedFrame::GstMappedFrame):
(WebCore::GstMappedFrame::~GstMappedFrame):
(WebCore::GstMappedFrame::get):
(WebCore::GstMappedFrame::componentData const):
(WebCore::GstMappedFrame::componentStride const):
(WebCore::GstMappedFrame::info):
(WebCore::GstMappedFrame::width const):
(WebCore::GstMappedFrame::height const):
(WebCore::GstMappedFrame::format const):
(WebCore::GstMappedFrame::planeData const):
(WebCore::GstMappedFrame::planeStride const):
(WebCore::GstMappedAudioBuffer::GstMappedAudioBuffer):
(WebCore::GstMappedAudioBuffer::~GstMappedAudioBuffer):
(WebCore::GstMappedAudioBuffer::get):
(WebCore::GstMappedAudioBuffer::info):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
(WebCore::GstMappedOwnedBuffer::create): Deleted.
(WebCore::GstMappedOwnedBuffer::~GstMappedOwnedBuffer): Deleted.
(WebCore::GstMappedFrame::GstMappedFrame): Deleted.
(WebCore::GstMappedFrame::~GstMappedFrame): Deleted.
(WebCore::GstMappedFrame::get): Deleted.
(WebCore::GstMappedFrame::ComponentData const): Deleted.
(WebCore::GstMappedFrame::ComponentStride const): Deleted.
(WebCore::GstMappedFrame::info): Deleted.
(WebCore::GstMappedFrame::width const): Deleted.
(WebCore::GstMappedFrame::height const): Deleted.
(WebCore::GstMappedFrame::format const): Deleted.
(WebCore::GstMappedFrame::planeData const): Deleted.
(WebCore::GstMappedFrame::planeStride const): Deleted.
(WebCore::GstMappedAudioBuffer::GstMappedAudioBuffer): Deleted.
(WebCore::GstMappedAudioBuffer::get): Deleted.
(WebCore::GstMappedAudioBuffer::info): Deleted.
(WebCore::GstMappedAudioBuffer::~GstMappedAudioBuffer): Deleted.
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrame::copyTo):
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoFrameLibWebRTC.cpp:
(WebCore::GStreamerVideoFrameLibWebRTC::ToI420):

Canonical link: <a href="https://commits.webkit.org/279191@main">https://commits.webkit.org/279191@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65736854913bcb0ee220d9971e537467ea3f0865

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52779 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32116 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5264 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56057 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3502 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55084 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38794 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3224 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/42844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2258 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54877 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/29831 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45558 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23957 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/26924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2873 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1661 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/48800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3026 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57649 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27918 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/3014 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/50242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29140 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45752 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/49520 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11524 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30056 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28892 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -546,6 +546,158 @@ uint64_t toGstUnsigned64Time(const MediaTime& mediaTime)
+    GstAudioInfo info;
+
+    if (!gst_audio_info_from_caps(&info, gst_sample_get_caps(sample.get()))) {
+        m_isValid = false;
```
This should be initialized to false, there should be no need to do this.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -546,6 +546,158 @@ uint64_t toGstUnsigned64Time(const MediaTime& mediaTime)
+{
+    if (m_isValid)
+        gst_audio_buffer_unmap(&m_buffer);
+    m_isValid = false;
```
You can move this inside the if

philn replied:
Let's make an early return.

philn committed:
[GStreamer] Move most GStreamerCommon classes implemenations to cpp file
https://bugs.webkit.org/show_bug.cgi?id=274524

Reviewed by Xabier Rodriguez-Calvar.

The GStreamerCommon.h file is included in many other files, so any change there triggers ~950
compile jobs on an incremental build. The template classes were left in the header, moving them to
the cpp file would increase verbosity too much. Also driving-by a couple GstMappedFrame methods were
renamed, adopting the camelCase style.

* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::GstMappedOwnedBuffer::create):
(WebCore::GstMappedOwnedBuffer::~GstMappedOwnedBuffer):
(WebCore::GstMappedOwnedBuffer::createSharedBuffer):
(WebCore::GstMappedFrame::GstMappedFrame):
(WebCore::GstMappedFrame::~GstMappedFrame):
(WebCore::GstMappedFrame::get):
(WebCore::GstMappedFrame::componentData const):
(WebCore::GstMappedFrame::componentStride const):
(WebCore::GstMappedFrame::info):
(WebCore::GstMappedFrame::width const):
(WebCore::GstMappedFrame::height const):
(WebCore::GstMappedFrame::format const):
(WebCore::GstMappedFrame::planeData const):
(WebCore::GstMappedFrame::planeStride const):
(WebCore::GstMappedAudioBuffer::GstMappedAudioBuffer):
(WebCore::GstMappedAudioBuffer::~GstMappedAudioBuffer):
(WebCore::GstMappedAudioBuffer::get):
(WebCore::GstMappedAudioBuffer::info):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
(WebCore::GstMappedOwnedBuffer::create): Deleted.
(WebCore::GstMappedOwnedBuffer::~GstMappedOwnedBuffer): Deleted.
(WebCore::GstMappedFrame::GstMappedFrame): Deleted.
(WebCore::GstMappedFrame::~GstMappedFrame): Deleted.
(WebCore::GstMappedFrame::get): Deleted.
(WebCore::GstMappedFrame::ComponentData const): Deleted.
(WebCore::GstMappedFrame::ComponentStride const): Deleted.
(WebCore::GstMappedFrame::info): Deleted.
(WebCore::GstMappedFrame::width const): Deleted.
(WebCore::GstMappedFrame::height const): Deleted.
(WebCore::GstMappedFrame::format const): Deleted.
(WebCore::GstMappedFrame::planeData const): Deleted.
(WebCore::GstMappedFrame::planeStride const): Deleted.
(WebCore::GstMappedAudioBuffer::GstMappedAudioBuffer): Deleted.
(WebCore::GstMappedAudioBuffer::get): Deleted.
(WebCore::GstMappedAudioBuffer::info): Deleted.
(WebCore::GstMappedAudioBuffer::~GstMappedAudioBuffer): Deleted.
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrame::copyTo):
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoFrameLibWebRTC.cpp:
(WebCore::GStreamerVideoFrameLibWebRTC::ToI420):

Canonical link: https://commits.webkit.org/279191@main
